9_collaborator_data/src/subset_sntemp_preds.R

```diff
@@ -1,24 +1,21 @@
 
 
-subset_sntemp_preds = function(ind_file,
+subset_sntemp_preds = function(out_file,
                                sub_net_file,
                                full_data_ind,
                                gd_config = 'lib/cfg/gd_config.yml'){
 
-  stream_temp_intermediates_wide = feather::read_feather(sc_retrieve(full_data_ind))
+  stream_temp_intermediates_wide <- feather::read_feather(sc_retrieve(full_data_ind))
 
   # subset set of seg_id_nats
-  sub_net = read.csv(sub_net_file)
-  sub_net_sites = as.character(unique(c(sub_net$from_reach, sub_net$to_reach)))
+  sub_net <- read_csv(sub_net_file, col_types='ccdd')
+  sub_net_sites <- unique(sub_net$seg_id_nat)
 
-  stream_temp_intermediates_wide_sub = stream_temp_intermediates_wide %>%
-    dplyr::filter(seg_id_nat %in% sub_net_sites)
+  stream_temp_intermediates_wide_sub <- stream_temp_intermediates_wide %>%
+    dplyr::filter(seg_id_nat %in% sub_net_sites) %>%
+    mutate(seg_tave_upstream = ifelse(seg_tave_upstream == -98.9, NA, seg_tave_upstream)) # convert numeric flag to NA
```

Comment:
Maybe add the comment that -98.9 indicates that a segment doesn't have any flow for given timestep. Since headwater segments do not have upstream segments (i.e. no flow upstream) many will have the -98.9 flag. See PRMS-SNTemp code here: https://github.com/nhm-usgs/prms/blob/2ca4e2d298fa853bcc87f28343cedceaad30ef27/prms/stream_temp.f90#L905-L907

Reply:
Good idea to explain...and should actually replace with 0 rather than NA, if the flag always means no flow?

Reply:
I think NA would be better because 0's would create artificially low basin temperatures when aggregating the segments

Reply:
Oh, whoops! You're right, of course. I started thinking this was a flow variable, which it clearly is not.

Reply:
Comment is added